### PR TITLE
Create a Kinesis Firehose module

### PR DESCRIPTION
**Pull request by amit-mazor**
### Creating a Firehose terraform module 
The module includes : 
* the creation of the required iam roles both for firehose and for the cloudwatch metrics streams
* the creation of s3 bucket for firehose backup
* the creation of log group per firehose delivery stream
* the option to create only firehose without a metrics stream 

TO DO :
Allow to use an existing bucket, and not only create a new one, 
its useful when creating multiple firehose streams and we dont want to have multiple separated buckets.

### CODE REVIEW

**Review by oded-dd — CHANGES_REQUESTED** [review]
Starting to look really good

File: examples/firehose/selected-metrics-cloudwatch-firehouse/variables.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+variable "include_all_namespaces" {
+  type = bool
+  description = "If set to true, the CloudWatch metric stream will include all available namespaces"
+  default = false
+}
+
+variable "include_metric_stream_namespaces" {
+  description = "List of specific namespaces to include in the CloudWatch metric stream, see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/aws-services-cloudwatch-metrics.html"
+  type        = list(string)
+  default     = ["EC2", "DynamoDB"]
```
[the remark this answers is not in the document]
Comment (amit-mazor) [reply examples/firehose/selected-metrics-cloudwatch-firehouse/variables.tf:28]: but its an example I want to show how to include only a few selected namespaces

File: examples/firehose/selected-metrics-cloudwatch-firehouse/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+## Usage
+
+In this example you need to configure the following variables:
+* firehose_stream --> The name of the Firehose delivery stream
+* endpoint_url --> The url of the Coralogix endpoint, see [Coralogix Endpoints](https://github.com/coralogix/terraform-coralogix-aws/blob/master/modules/firehose/README.md)
+* privatekey --> Coralogix account logs privatekey
+Since the privatekey is private and we cant put it hardcoded, it can be exported instead of insert it as an input each time:
+export TF_VAR_coralogix_privatekey="your-coralogix-private-key"
+* include_all_namespaces --> this variable needs to be set to 'false' to disable the creation of all available cloudwatch namespaces
+* include_metric_stream_namespaces --> The list of the the desired namespaces, for example: ["EC2", "DynamoDB"]. For the full list of the available namespaces and how they need to be mentioned, please see [namespaces list](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/aws-services-cloudwatch-metrics.html)'
```
[the remark this answers is not in the document]
Comment (amit-mazor) [reply examples/firehose/selected-metrics-cloudwatch-firehouse/README.md:21]: Our docs are not related to these namespaces that AWS allow us to use, 
the goal of this link is to show the users how to select namespaces, our docs dont have this list

File: examples/firehose/full-cloudwatch-firehose/main.tf [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  backend "local" {
+    path = "./terraform.tfstate"
+  }
+}
+
+module "cloudwatch_firehose_coralogix" {
+  source                           = "../../../modules/firehose"
+  firehose_stream                  = var.coralogix_firehose_stream_name
+  endpoint_url                     = var.coralogix_endpoint_url
+  privatekey                       = var.coralogix_privatekey
```
[the remark this answers is not in the document]
Comment (amit-mazor) [reply examples/firehose/full-cloudwatch-firehose/main.tf:11]: its true by default so I just deleted the var from here

File: examples/firehose/selected-metrics-cloudwatch-firehouse/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Firehose Delivery Stream with CloudWatch metrics stream including specific metrics namespaces 
+Configuration in this directory creates a firehose delivery stream, and a CloudWatch metrics stream. 
+The CloudWatch metrics stream in this example includes only specific namespaces that are inserted in the list.   
+All of the namespaces are created under one metric stream, for example:
+```
+coralogix-metric-stream:
+AWS/EC2
+AWS/DynamoDB
+AWS/EBS
+```
```
[the remark this answers is not in the document]
Comment (amit-mazor) [reply examples/firehose/selected-metrics-cloudwatch-firehouse/README.md:11]: added the aws docs link, and kept this list as an example

**Comment by amit-mazor** [comment]
@oded-dd @avivgold098
* Updated all of the examples - changed the source, the defaults, and of the repeated comments.
* Also, added a new example for deploying multiple firehoses.  
* About the var of the endpoint url, im trying to update it to be based on the region input instead of the full url.

TO DO:
* Add readme
* Add changelolg

**Review by avivgold098 — APPROVED** [review]

**Pull request merged**